gadget/install/partition.go

```diff
@@ -70,10 +70,10 @@ func createMissingPartitions(dl *gadget.OnDiskVolume, pv *gadget.LaidOutVolume)
 }
 
 // removeCreatedPartitions removes partitions added during a previous install.
-func removeCreatedPartitions(dl *gadget.OnDiskVolume) error {
+func removeCreatedPartitions(lv *gadget.LaidOutVolume, dl *gadget.OnDiskVolume) error {
 	indexes := make([]string, 0, len(dl.Structure))
 	for i, s := range dl.Structure {
-		if s.CreatedDuringInstall {
+		if gadget.WasCreatedDuringInstall(lv, s) {
```

Comment:
We should skip the ubuntu-save here, perhaps with a log message that we are doing so. Maybe a TODO:UC20 then?

The may still be a scenario in which ubuntu-save is created, encrypted, but for some reason unusable (fs/LUKS corruption or whatnot). In such scenario, we will leave the partition there for now, but a reinstall should probably have a way of indicating that it should wipe everything.

Reply:
Note that there could also be devices that need to have all their partitions created at image flash time, so we might have an unencrypted/empty ubuntu-save around on first install but I think we can handle this at a later date when we decide how to distinguish factory reset vs actually factory first install

Reply:
I think we should keep install removing everything as first step and do what we can/necessary changes when we have landed more of save support.